scripts/hyperkit.js

```diff
@@ -0,0 +1,20 @@
+// This downloads hyperkit, builds it, and puts the binary in the right place.
+
+const fs = require('fs');
+const os = require('os');
+const { spawn, spawnSync } = require('child_process');
+
+fs.mkdirSync("./resources/" + os.platform(), { recursive: true });
+
+// The version of hyperkit to build
+let ver = "v0.20210107";
+
+// TODO: rewrite to remove callbackitis
+spawn('git', ['clone', '--depth', '1', '--branch', ver, "https://github.com/moby/hyperkit.git"], { cwd: '/tmp/' }).on('exit', () => {
```

Comment:
Any particular reason we need to use git (pulling in all the history), rather than just the [archive](https://github.com/moby/hyperkit/archive/v0.20210107.tar.gz)?  Or use a git submodule?

Reply:
@mook-as the make script requires git in order to build the version into the binary. minikube checks the version and provides an error if not there. This was done in a way to get that version into the built binary.

Reply:
Ah, in that case we might want to add a comment here (so that we don't try to "optimize" it in the future and drop the git part).

Reply:
Alternatively, we could do `make GIT_VERSION=${ver} GIT_VERSION_SHA1=…` to make it embed an arbitrary version string.  (It looks like we _do_ need to have that `GIT_VERSION_SHA1` because there's an [assertion for it](https://github.com/moby/hyperkit/blob/40cbd5caa9580e2b6de8fef53f4581943c430312/src/lib/pci_virtio_net_vpnkit.c#L324).)

Reply:
I took the path of adding docs. The context is out of my head now.